Source/WebCore/svg/SVGLength.h

```diff
@@ -48,7 +48,8 @@ class SVGLength : public SVGValueProperty<SVGLengthValue> {
         SVG_LENGTHTYPE_MM           = static_cast<unsigned>(SVGLengthType::Millimeters),
         SVG_LENGTHTYPE_IN           = static_cast<unsigned>(SVGLengthType::Inches),
         SVG_LENGTHTYPE_PT           = static_cast<unsigned>(SVGLengthType::Points),
-        SVG_LENGTHTYPE_PC           = static_cast<unsigned>(SVGLengthType::Picas)
+        SVG_LENGTHTYPE_PC           = static_cast<unsigned>(SVGLengthType::Picas),
+        SVG_LENGTHTYPE_REMS           = static_cast<unsigned>(SVGLengthType::Rems)
```

Comment:
I believe this will be incorrect. `rem` likely needs to return SVG_LENGTHTYPE_UNKNOWN, since it is not defined as part of the IDL (https://www.w3.org/TR/SVG2/types.html#InterfaceSVGLength).

This WPT test seems to imply the same - https://github.com/web-platform-tests/wpt/blob/85b598e652/svg/types/scripted/SVGLength-rem.html.

Reply:
Similar from https://github.com/WebKit/WebKit/pull/22653

https://svgwg.org/svg2-draft/types.html#InterfaceSVGLength
> If other types of lengths are supported and used, the SVGLength uses the SVG_LENGTHTYPE_UNKNOWN unit type. See below for details on how the other properties of an SVGLength operate with these types of lengths.

to add to the commit message.

Reply:
Also the SVG WG has been rechartered recently and a work of maintenance on the specification will happen soon or later (no new features) just streamlining the spec to make sure it describes what is currently implemented where it makes sense.

Reply:
I am not sure - this is needed since I did this changed - https://github.com/WebKit/WebKit/commit/e552416b237a8358fc0acbeff8315263139be4a3#diff-c71404f0fc55aa5cae1cf06de7032bc57919bdc529cd4bcac1d430f5e30e4bcaR84

```

     {
         // Per spec: https://svgwg.org/svg2-draft/types.html#__svg__SVGLength__SVG_LENGTHTYPE_UNKNOWN
         if (m_value.lengthType() > SVGLengthType::Picas)
             return 0;
         return static_cast<unsigned>(m_value.lengthType());
     }
```

So technically, it will be treated as `Unknown`.

Reply:
@Ahmad-S792 i was curious why `unitType` was returning 0 even after changing it to a proper value in this file and this check wasn't immediately apparent. do we really need a runtime guard for this? i'd think test failures would be a good indication of not following the spec?

Reply:
@nullhook - From IDL, we only have '10; values till SVGLengthType::Picas and when we add anything new, it should return `0` - which is SVGLengthType. For all new length types WPT - if you don't do this. They will fail one of subtests which is if the value is 13, it should return '0'. Without this, it would return '13' and will fail WPT.

Reply:
this is not logical in current STP
https://codepen.io/webcompat/pen/gbOKXxX

`rem` is an invalid value as it is supposed to not be known but still it positionned it correctly.

![Screenshot 2025-03-24 at 10 47 06](https://github.com/user-attachments/assets/b66d104a-7579-4c22-abff-1afcb28681f1)

Reply:
pushed another commit to reflect layout and DOMInterface's SVGLength properties

![image](https://github.com/user-attachments/assets/83568669-eacb-4ec6-a507-90f7fa5d03b3)